commerce-subscription-web/src/main/java/com/liferay/commerce/subscription/web/internal/display/context/CPDefinitionSubscriptionInfoDisplayContext.java

```diff
@@ -85,6 +99,48 @@ public PortletURL getPortletURL() throws PortalException {
 		return portletURL;
 	}
 
+	public boolean hasRecurringPaymentMethod() throws PortalException {
+		List<CommerceChannelRel> commerceChannelRels =
+			CommerceChannelRelLocalServiceUtil.getCommerceChannelRels(
+				CPDefinition.class.getName(), getCPDefinitionId(), -1, -1,
+				null);
+
+		for (CommerceChannelRel commerceChannelRel : commerceChannelRels) {
+			boolean channelHasRecurringPaymentMethod = false;
+
+			CommerceChannel commerceChannel =
+				commerceChannelRel.getCommerceChannel();
+
+			List<CommercePaymentMethodGroupRel> commercePaymentMethodGroupRels =
+				_commercePaymentMethodGroupRelLocalService.
+					getCommercePaymentMethodGroupRels(
+						commerceChannel.getSiteGroupId(), true);
+
+			for (CommercePaymentMethodGroupRel commercePaymentMethodGroupRel :
+					commercePaymentMethodGroupRels) {
+
+				if (commercePaymentMethodGroupRel.isActive()) {
+					CommercePaymentMethod commercePaymentMethod =
+						_commercePaymentMethodRegistry.getCommercePaymentMethod(
+							commercePaymentMethodGroupRel.getEngineKey());
+
+					if (commercePaymentMethod.isProcessRecurringEnabled()) {
+						channelHasRecurringPaymentMethod = true;
+					}
+				}
+			}
+
+			if (!channelHasRecurringPaymentMethod) {
+				return false;
+			}
```

Comment:
Is this the logic we want? This seems to be saying:
 > Get all channels that publish this product. If any of those channels doesn't have a recurring payment method, then for this product `hasRecurringPaymentMethod=false`

I think I would expect:
 > Get all channels that publish this product. If any of those channels has a recurring payment method, then for this product `hasRecurringPaymentMethod=true`
Can you confirm that it's correct as-is?

Reply:
Oh interesting, just saw:
> all-channels-associated-with-this-product-must-have-at-least-one-payment-method-active-that-supports-recurring-payments

Guess that is what we want :)